src/util/nondet.cpp

```diff
@@ -60,30 +38,35 @@ symbol_exprt generate_nondet_int(
   // Constrain the non deterministic integer with a lower bound of `min_value`.
   //   ASSUME(name_prefix::nondet_int >= min_value)
   instructions.add(
-    code_assumet(
-      binary_predicate_exprt(
-        nondet_symbol, ID_ge, from_integer(min_value, int_type))));
+    code_assumet(binary_predicate_exprt(nondet_symbol, ID_ge, min_value_expr)));
 
   // Constrain the non deterministic integer with an upper bound of `max_value`.
   //   ASSUME(name_prefix::nondet_int <= max_value)
   instructions.add(
-    code_assumet(
-      binary_predicate_exprt(
-        nondet_symbol, ID_le, from_integer(max_value, int_type))));
+    code_assumet(binary_predicate_exprt(nondet_symbol, ID_le, max_value_expr)));
 
   return nondet_symbol;
 }
 
-/// Pick nondeterministically between imperative actions 'switch_cases'.
-/// \param name_prefix: Name prefix for fresh symbols (should be function id)
-/// \param switch_cases: List of codet objects to execute in each switch case.
-/// \param int_type: The type of the int used to non-deterministically choose
-///   one of the switch cases.
-/// \param mode: Mode (language) of the symbol to be generated.
-/// \param source_location: The location to mark the generated int with.
-/// \param symbol_table: The global symbol table.
-/// \return Returns a nondet-switch choosing between switch_cases. The resulting
-///   switch block has no default case.
+symbol_exprt generate_nondet_int(
+  const mp_integer &min_value,
+  const mp_integer &max_value,
+  const std::string &basename_prefix,
+  const typet &int_type,
+  const source_locationt &source_location,
+  allocate_objectst &allocate_objects,
+  code_blockt &instructions)
+{
+  PRECONDITION(min_value < max_value);
```

Comment:
`<=` is fine

Reply:
Done (in a new commit, as the previous version used `<` too).